src/util/misc.ts

```diff
@@ -161,3 +162,118 @@ export function dedup<T>(arr: T[]): T[] {
 
     return res;
 }
+
+/**
+ * Build a flattened version of the array of arrays `arr` and return it.
+ */
+export function flatten<T>(arr: Iterable<T[]>): T[] {
+    const res: T[] = [];
+    for (const el of arr) {
+        res.push(...el);
+    }
+
+    return res;
+}
+
+/**
+ * Given a list of T's `things` and a partial ordering between them `order` return
+ * a topologically sorted version of `things`. For any pair `[a,b]` in `order` we assume
+ * that `a` has to come before `b`.
+ *
+ * @param things
+ * @param relation
```

Comment:
These line can be removed as do not provide anything useful.

Reply:
Fixed in dcf955c.